tests/test_products.py

```diff
@@ -0,0 +1,989 @@
+        expected_bands_names = [v[0] for v in expected_bands_mapping.values()]
+        expected_value = (132.0 - 122.0) / (132.0 + 122.0)
+        self.do_test_product_for_macro(NDVI750(), expected_bands_mapping,
+                                       expected_bands_names, expected_value)
+
+    def test_for_macro_with_no_data(self):
+        product = self.do_test_for_macro_with_no_data(NDVI750())
+        raster = product.raster
+        self.assertTrue(raster.image.mask[0, 1, 2])
+        self.assertFalse(raster.image.mask[0, 2, 3])
+        self.assertFalse(raster.image.mask[0, 0, 0])
+        self.assertFalse(raster.image.mask[0, 1, 3])
+
+
+class TestLandCoverIndex(unittest.TestCase):
+
+    def test_product_for_macro(self):
+        product_generator = LandCoverIndex()
+        expected_bands_mapping = {
+            'red': ['HC_830'],
+            'green': ['HC_690'],
+            'blue': ['HC_550']
+        }
+        expected_bands_names = [v[0] for v in expected_bands_mapping.values()]
+        product = product_generator.apply(sensor_bands_info(), macro_raster(), metadata=True)
+        self.assertEqual(product.name, product_generator.name)
+        self.assertEqual(product.display_name, product_generator.display_name)
+        self.assertEqual(product.description, product_generator.description)
+        self.assertEqual(product.min, product_generator.min)
+        self.assertEqual(product.min, product_generator.min)
+        self.assertEqual(product.required_bands, product_generator.required_bands)
+        self.assertEqual(product.output_bands, product_generator.output_bands)
+        self.assertEqual(product.unit, product_generator.unit)
+        self.assertCountEqual(product.bands_mapping, expected_bands_mapping)
+        self.assertCountEqual(product.used_bands, expected_bands_names)
+        raster = product.raster
+        self.assertTrue((raster.bands_data('red') == 138.0).all())
+        self.assertTrue((raster.bands_data('green') == 120.0).all())
+        self.assertTrue((raster.bands_data('blue') == 106.0).all())
+
+    def test_match_bands_for_legend(self):
+        product_generator = LandCoverIndex()
+        expected_bands_mapping = {
+            'red': ['HC_830'],
+            'green': ['HC_690'],
+            'blue': ['HC_550']
+        }
+        bands_mapping = product_generator.match_bands_for_legend(sensor_bands_info(), macro_raster().band_names)
+        self.assertCountEqual(bands_mapping, expected_bands_mapping)
+
+
+class TestRGBEnhanced(unittest.TestCase):
+    enhanced_bands = ['red_enhanced', 'green_enhanced', 'blue_enhanced']
+
+    @classmethod
+    def enhanced_raster(cls):
+        source_raster = make_test_raster(142, cls.enhanced_bands, dtype=np.uint8)
+        for i, band in enumerate(cls.enhanced_bands):
+            source_raster.image.data[i, :, :] = 2 * i + 100
+        return source_raster
+
+    def test_product(self):
+        product_generator = ProductsFactory.get_object('rgbenhanced')
```

Comment:
is rgbenhanced is a product or just raster with rgb enhanced bands with  a colormap visualization?

and if it is a product it should be located in our internal libraries

Reply:
I don't understand the comment